tests/e2e/test.bats

```diff
@@ -41,40 +41,39 @@ fi
       assert_match "k8s:enc:kms:v1:azurekmsprovider" "${output}"
       assert_success
     fi
-
-    # cleanup
-    run kubectl delete secret secret1 -n default
 }
 
 @test "check if metrics endpoint works" {
-    # ToDo - enbale this test after v0.0.12 release
-    if [ ${SKIP_METRICS} = true ]; then
-      skip "metrics endpoint is not yet released in soak cluster."
-    fi
-
-    kubectl run curl --image=curlimages/curl:7.75.0 -- tail -f /dev/null
-    kubectl wait --for=condition=Ready --timeout=60s pod curl
+    local randomString=$(openssl rand -hex 5)
```

Comment:
Maybe make the pod name a local variable `local curl_pod_name=curl-$(openssl rand -hex 5)` and use that within the scope of the test?